float64 mean_x
float64 mean_y
float64 var_x
float64 var_y
float64 var_xy
float64 weight
